### PR TITLE
feature/POSTs

### PR DESCRIPTION
**Pull request by theeleking**
Created POST single and multiple functions for activities and GET functions for Food. Also fixed type checking for glucose conversion.

### CODE REVIEW

File: test/gb/food.test.ts [hunk 1]
```diff
@@ -21,19 +409,202 @@ describe('with mocked food get call', () => {
...
+            timestamp : 1622832265000,
+            carbohydrates : 400,
+            fibers: 34,
+            description: 'desc'
+        },{
+            timestamp : 1622832285000,
+            carbohydrates : 400,
+            fibers: 34,
+            description: 'desc'
+        }];
```
Comment (RinseV) [inline test/gb/food.test.ts:424]: Also make a test case for a response that does not include any carbohydrates, so you can see whether the error is thrown correctly

File: test/gb/insulin.test.ts [hunk 2]
```diff
@@ -16,24 +24,305 @@ describe('with mocked insulin get call', () => {
...
+                    Authorization: 'Bearer testToken'
+                }),
+                data: [POSTData1, POSTData2]
+            })
+        );
+    });
+});
+
+describe('convert response to models', () => {
+    test('convert single response to single model', () => {
```
Comment (RinseV) [inline test/gb/insulin.test.ts:179]: This is only testing RAPID insulin type, SHORT is untested

File: test/gb/mood.test.ts [hunk 3]
```diff
@@ -61,12 +59,8 @@ describe('with mocked moods get call', () => {
```
Comment (RinseV) [inline test/gb/mood.test.ts:59]: Test cases are missing for ``convertMoodResponseToModel()`` though that should have been caught sooner

**Review by RinseV — CHANGES_REQUESTED** [review]
Looks good, some comments are inconsistent for ``food.ts``, also missing some test cases for ``food`` and ``insulin``, please see test coverage to see what is missing.

File: src/gb/objects/food.ts [hunk 4]
```diff
@@ -1,21 +1,256 @@
...
+        //console.log(response)
+        //console.log(activities)
+        activities.forEach((activity: ActivityModel) => {
+            const key =
+                Object.keys(FoodPropertyKeys).find(
+                    (key) => FoodPropertyKeys[key] === activity.property.translationKey
+                ) ?? activity.property.translationKey; //used the online key if no translation is found
+            model[key] = activity.value;
+        });
+        if (model.carbohydrates === -1) {
```
[the remark this answers is not in the document]
Comment (theeleking) [reply src/gb/objects/food.ts:207]: -1 checking was removed in favor of setting it to 0

File: test/gb/food.test.ts (hunk 1 above)
Comment (theeleking) [reply test/gb/food.test.ts:422]: -1 checking was removed in favor of setting it to 0, so no test is needed for checking errors

**Review by RinseV — APPROVED** [review]
👍

**Pull request merged**